Tasks/AzureRmWebAppDeployment/AzureRmWebAppDeployment.ts

```diff
@@ -37,6 +39,20 @@ async function run() {
         if (!tl.exist(webDeployPkg)) {
             throw new Error(tl.loc('Invalidwebapppackageorfolderpathprovided', webDeployPkg));
         }
+
+        var isParamFilePresentInPackage = false;
+        var isFolderBasedDeployment = !msDeployUtility.fileExists(webDeployPkg);
+        
+        if (!isFolderBasedDeployment) {
+            isParamFilePresentInPackage = await msDeployUtility.containsParamFile(webDeployPkg);
+        }
+
+        if(!tl.filePathSupplied('SetParametersFile')) {
+            setParametersFile = null;
+        }
+        else if (!msDeployUtility.fileExists(setParametersFile)) {
+            throw Error(tl.loc('SetParamFilenotfound0', setParametersFile));
+        }
```

Comment:
Why am I seeing this code as addition this code is already not checked into master by Vincent